diffyscan/utils/binary_verifier.py

```diff
@@ -0,0 +1,142 @@
+    )
+    logger.info(
+        f'{green("0000 00 STOP")} - the actual bytecode instruction exists when the expected doesn\'t'
+    )
+    logger.divider()
+    for i in range(len(checkpointsArray)):
+        currInd = checkpointsArray[i]
+        prevInd = checkpointsArray[i - 1] if i > 0 else None
+        if prevInd and prevInd != currInd - 1:
+            print("...")
+
+        actual = (
+            actualInstructions[currInd] if currInd < len(actualInstructions) else None
+        )
+        expected = (
+            expectedInstructions[currInd]
+            if currInd < len(expectedInstructions)
+            else None
+        )
+
+        if not actual and expected:
+            params = "0x" + expected["bytecode"][2:]
+            print(
+                logger.red(
+                    f'{to_hex(currInd, 4)} {to_hex(expected["op"]["code"])} {expected["op"]["name"]} {params}'
+                )
+            )
+        elif actual and not expected:
+            params = "0x" + actual["bytecode"][2:]
+            print(
+                logger.green(
+                    f'{to_hex(currInd, 4)} {to_hex(actual["op"]["code"])} {actual["op"]["name"]} {params}'
+                )
+            )
+        elif actual and expected:
+            opcode = (
+                to_hex(actual["op"]["code"])
+                if actual["op"]["code"] == expected["op"]["code"]
+                else bgRed(to_hex(actual["op"]["code"]))
+                + " "
+                + bgGreen(to_hex(expected["op"]["code"]))
+            )
+            opname = (
+                actual["op"]["name"]
+                if actual["op"]["name"] == expected["op"]["name"]
+                else bgRed(actual["op"]["name"]) + " " + bgGreen(expected["op"]["name"])
+            )
+            actualParams = (
+                "0x" + actual["bytecode"][2:] if len(actual["bytecode"]) > 2 else ""
+            )
+            expectedParams = (
+                "0x" + expected["bytecode"][2:] if len(expected["bytecode"]) > 2 else ""
+            )
+
+            paramsLength = len(expected["bytecode"]) // 2 - 1
+            isImmutable = immutables.get(expected["start"] + 1) == paramsLength
+            params = (
+                actualParams
+                if actualParams == expectedParams
+                else (
+                    bgYellow(actualParams) + " " + bgGreen(expectedParams)
+                    if isImmutable
+                    else bgRed(actualParams) + " " + bgGreen(expectedParams)
+                )
+            )
+            print(f"{to_hex(currInd, 4)} {opcode} {opname} {params}")
+        else:
+            raise ValueError("Invalid bytecode difference data")
+
+
+def parse(bytecode):
+    buffer = bytes.fromhex(bytecode[2:] if bytecode.startswith("0x") else bytecode)
+    instructions = []
+    i = 0
+    while i < len(buffer):
+        opcode = buffer[i]
+        length = 1 + (opcode - 0x5F if 0x5F <= opcode <= 0x7F else 0)
```

Comment:
why `0x5F` and `0x7F` aren't constants?

Reply:
Plus, it is worth adding a check that the actual length is enough for PUSH(X), wdyt?